scalableminds/webknossos
# ome-ngff header for zarr streaming

- Added an endpoint to request the OME-NGFF Header for a specific layer and a `OmeNgffHeader` Class
    - Note that the code assumes that the path to a mag is always in the shortened form on disk (if it is possible)
    - Note that the `OmeNgffHeader` class doesn't implement all fields from the standard but only the ones that we currently need
- Rewrite part of the .zarray route to use the existing `ZarrHeader` class

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Send a request to `GET /data/zarr/:organization/:dataset/:layer/.zattrs`, e.g. `GET /data/zarr/sample_organization/l4_sample/color/.zattrs`
- you should get back a Json that looks similar to this one:
```
{
    "multiscales": [{
        "version": "0.4",
        "name": "color",
        "axes": [{
            "name": "c",
            "type": "channel"
        }, {
            "name": "x",
            "type": "space",
            "unit": "nanometer"
        }, {
            "name": "y",
            "type": "space",
            "unit": "nanometer"
        }, {
            "name": "z",
            "type": "space",
            "unit": "nanometer"
        }],
        "datasets": [{
            "path": "1",
            "coordinateTranformations": [{
                "type": "scale",
                "scale": [1, 11.239999771118164, 11.239999771118164, 28]
            }]
        }, {
            "path": "2-2-1",
            "coordinateTranformations": [{
                "type": "scale",
                "scale": [1, 22.479999542236328, 22.479999542236328, 28]
            }]
        }, {
            "path": "4-4-1",
            "coordinateTranformations": [{
                "type": "scale",
                "scale": [1, 44.959999084472656, 44.959999084472656, 28]
            }]
        }, {
            "path": "8-8-2",
            "coordinateTranformations": [{
                "type": "scale",
                "scale": [1, 89.91999816894531, 89.91999816894531, 56]
            }]
        }, {
            "path": "16-16-4",
            "coordinateTranformations": [{
                "type": "scale",
                "scale": [1, 179.83999633789062, 179.83999633789062, 112]
            }]
        }]
    }]
}
```

### Issues:
- fixes #6206 

------
